src/SmartComponents/ActivityTable/__tests__/ActivityTable.tests.js

```diff
@@ -133,13 +133,13 @@ describe('ActivityTable', () => {
 
     await waitFor(() => {
       expect(fetchExecutedTasks).toHaveBeenCalled();
-      userEvent.click(screen.getByLabelText('Conditional filter'));
-      userEvent.click(screen.getAllByText('Status')[0]);
-      userEvent.click(screen.getByLabelText('Options menu'));
-      userEvent.click(screen.getAllByText('Completed')[0]);
-      expect(screen.getByText('taskA')).toBeInTheDocument();
-      expect(screen.queryByText('taskB')).not.toBeInTheDocument();
     });
+    userEvent.click(screen.getByLabelText('Conditional filter'));
+    userEvent.click(screen.getAllByText('Status')[0]);
+    userEvent.click(screen.getByLabelText('Options menu'));
+    userEvent.click(screen.getAllByText('Completed')[0]);
```

Comment:
These should probably be wrapped in an `act`, as they are triggering component state updates. See https://testing-library.com/docs/react-testing-library/api#act.

There are also a few warnings regarding this when running the tests.

Reply:
This breaks everything now, and I don't understand why.

```
    await waitFor(() => {
      expect(fetchExecutedTasks).toHaveBeenCalled();
    });
    act(async () => {
      await userEvent.click(screen.getByLabelText('Conditional filter'));
      await userEvent.click(screen.getAllByText('Status')[0]);
      await userEvent.click(screen.getByLabelText('Options menu'));
      userEvent.click(screen.getAllByText('Running')[0]);
    });
    expect(screen.getByText('taskB')).toBeInTheDocument();
    expect(screen.queryByText('taskA')).not.toBeInTheDocument();
    ```
    
If I don't add the `await`s, then it can't find `Options menu`, but when I add the `await`s, it causes the entire test suite to fail.

Reply:
@johnsonm325 I'm not sure this is correctly used. you should not need to pass act and `async` function nor should you need to await the `.click`s. 

It also doesn't seem right to have the `waitFor` above, if the clicks should trigger the the fetch. Or if the clicks trigger another fetch there should be another `waitFor`.